README.md

```diff
@@ -113,20 +113,28 @@ For Android remote notifications, setting your sender ID and adding your `google
+| `protocol_repo`               | Github repo where the protocol definitions are located.                                 | `RADAR-Base/RADAR-aRMT-protocols`                                                                                |
+| `protocol_branch`             | Github branch where the protocol definitions should be read from                        | `master`                                                                                                         |
+| `protocol_path`               | Path inside a project name that should be read for a protocol                           | `protocol.json`                                                                                                  |
+| `kafka_specification_url`     | URL of the Kafka topic specification                                                    | <https://api.github.com/repos/RADAR-base/radar-schemas/contents/specifications/active/aRMT-1.4.3.yml?ref=master> |
+| `platform_instance`           | Title of RADAR Base / platform instance                                                 | `RADAR-CNS`                                                                                                      |
+| `participant_attribute_order` | Map that specifies the order in which the attributes are matched with the protocol path | `{Human-readable-identifier: -1}`                                                                                |
+
 #### Conditions
 
 Conditions can be added to remote config variables to target specific groups of users. Different condition rule types are supported: app, platform, country/region, user property, date/time, and random percentile. For example a `protocol_branch` config value can be different based on the user property `projectId`.
 
+#### Protocol Attributes
+
+A user/subject can have `attributes` which are taken from Management Portal. These could determine what protocol would be pulled for the user from Github ([RADAR aRMT Protocols](https://github.com/RADAR-base/RADAR-aRMT-protocols)). The repository should follow the format: `/PROJECT_NAME/ATTRIBUTE-KEY-1/ATTRIBUTE-VALUE-1/protocol.json`. Please note that a default `protocol.json` file must always be present in the project directory, e.g.: `/PROJECT_NAME/protocol.json`.
+
+For multiple attributes, you may set them as follows: `/PROJECT_NAME/ATTRIBUTE-KEY-1/ATTRIBUTE-VALUE-1/ATTRIBUTE-KEY-2/ATTRIBUTE-VALUE-2/[...]/protocol.json`. If multiple attribute keys are found in the same path (e.g. `/PROJECT_NAME/ATTRIBUTE-KEY-1/` and `/PROJECT_NAME/ATTRIBUTE-KEY-2/`), and these attributes are present for the user, the `participant_attribute_order` from the Remote Config will be used to determine which attribute takes precedence. If this is not present, the default value is `{Human-readable-identifier: -1}`; the human readable id will always be the highest priority. If an order value is not present for the attribute, the default value (`MAX_INT_VALUE`) would be used.
```

Comment:
> For multiple attributes, you may set them as follows

Is this for setting map type attributes? I think we should keep the structure simple with just one level of hierarchy like `/PROJECT_NAME/ATTRIBUTE-1-KEY/ATTRIBUTE-1-VALUE/`, `/PROJECT_NAME/ATTRIBUTE-2-KEY/ATTRIBUTE-2-VALUE/` and so on. if in the future we need to support map type attributes, they can be flattened in the redcap integration app before adding to MP

Reply:
Oh okay I see. Yes this is for map type attributes. Yeah flattening it would be good to make things less complicated. But the solution should support both since it just goes through the Github tree. Is that fine?

Reply:
I think maybe it's better to remove it. It will make the code concise and also decrease the load time of protocols in the app since it will stop at the first level. What do you think? I am good with both ways.

Reply:
Okay sure. I can remove the recursion.